### ISSUE TITLE
deprecate source.load, style.load, etc to simplify + clarify eventing

### ISSUE THREAD

**Issue by mollymerp** [mapbox/mapbox-gl-js#3970]
per chat w @lucaswoj yesterday and with added motivation from #3958, we should simply eventing by removing all events that do not fall within the schema of the `data` events from #3590, and extending the current implementation to cover the use cases of the old events.

**Comment by gertcuykens** [comment]
hmm I am unable to figure out how i should do the `style.load` event now? Tried `load` also.

example: https://gps.nyimbo.nl press [+] to load new style, event doesn't get fired and when pressed again I get warning?

(ps dont quote link please so I can remove later)

**Comment by lucaswoj** [comment]
@gertcuykens `style.load` was never part of our public API. I believe the [`styledata`](https://www.mapbox.com/mapbox-gl-js/api/#Map.event:styledata) event has the behaviour you're looking for.

**Comment by gertcuykens** [comment]
yes and no :) because when I use `styledata` instead i get allot of error messages in the form of
![image](https://cloud.githubusercontent.com/assets/863731/22385437/ee4d11d6-e4d2-11e6-9131-d725a85e75cb.png)

in previous versions using `style.load` i was guaranteed I could add new sources and build my map back up again when the event got fired when switching styles. Note, that I could have just been lucky things didn't break :) But as far as I can tell there is no other event that only get called once when switching styles?

```
map.on('styledata', function () {

  map.addSource('markers', {
    type: 'geojson',
    data: markers,
    cluster: true,
    clusterMaxZoom: 14,
    clusterRadius: 30
  })

  map.addSource('traces', {
    type: 'geojson',
    data: traces
  })

  if (location.search.substring(1) !== '') {
    draw()
  } else {
    map.getSource('traces').setData(traces)
  }
  map.getSource('markers').setData(markers)
  paint()
  threadBuffer.postMessage(null)
  threadSPOT.postMessage(null)

}
```

**Comment by lucaswoj** [comment]
Thanks for the heads-up about that caveat to `styledata` @gertcuykens. We will fix that in conjunction with deprecating `style.load`.

**Comment by ryanhamley** [comment]
Updating issue name since `source.load` is no longer an event
